devito/passes/clusters/aliases.py

```diff
@@ -699,12 +699,17 @@ def iter(self, ispace):
         items = []
         for alias, (intervals, aliaseds, distances) in self.items():
             mapper = {i.dim: i for i in intervals}
+            mapper.update({i.dim.parent: i for i in intervals
+                           if i.dim.is_NonlinearDerived})
 
+            # Becomes True as soon as a Dimension in `ispace` is found to
+            # be independent of `intervals`
+            flag = False
             writeto = []
-            for n, i in enumerate(ispace.intervals):
-                interval = mapper.get(i.dim)
-                if interval is not None:
-                    if not writeto and interval == interval.zero():
+            for i in ispace.intervals:
+                try:
```

Comment:
```
interval = mapper.get(i.dim)
if not interval:
    if any(i.dim in d._defines for d in mapper):
         # E.g., `i.dim` is `x0_blk0` in `x0_blk0[0,0]<0>`
          pass
     else:
           # E.g., `t[0,0]<0>` in the case of t-invariant aliases
          flag = True
```

would avoid large try/except.

Reply:
Hang on, I'll try to make it nicer

Reply:
now?

Reply:
but do we really need the try/except?

Reply:
it's completely pythonic and I'm sure we use the same pattern somewhere else in Devito

so: "really need" -> no , "is it fine to have" -> yes

Reply:
ok gtg